### PR TITLE
Fix a bug with relational mapping tests set with the incorrect database

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Fixes https://github.com/finos/legend-studio/issues/649
## How did you test this change?

https://user-images.githubusercontent.com/73408381/140786704-66053ac3-dbb9-4258-abfb-2c4623d02e37.mov

### CODE REVIEW

**Comment by akphi** [comment]
@YannanGao-gs Could you elaborate on the context of this bug, what are you aiming to fix?

File: .changeset/beige-humans-develop.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+'@finos/legend-studio': patch
+---
```
Comment (akphi) [inline .changeset/beige-humans-develop.md:3]: Properly should add a changeset here, this is technically a breaking change.

File: packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts [hunk 2]
```diff
@@ -520,13 +519,13 @@ export class MappingTestState {
...
       this.setInputDataState(newInputDataState);
-    } else if (source instanceof Table || source instanceof View) {
+    } else if (source instanceof TableAlias) {
       const newInputDataState = new MappingTestRelationalInputDataState(
         this.editorStore,
         this.mappingEditorState.mapping,
         new RelationalInputData(
           PackageableElementExplicitReference.create(
-            guaranteeNonNullable(source.schema.owner),
+            guaranteeNonNullable(source.relation.ownerReference.value),
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts:528]: is the `guaranteeNonNullable` necessary here?

File: packages/legend-studio/src/components/editor/edit-panel/mapping-editor/InstanceSetImplementationSourceSelectorModal.tsx [hunk 3]
```diff
@@ -131,11 +135,21 @@ export const InstanceSetImplementationSourceSelectorModal = observer(
...
-          ),
-        ),
+        editorStore.graphManagerState.graph.ownDatabases
+          .flatMap((e) =>
+            e.schemas.flatMap((schema) =>
+              (schema.tables as (Table | View)[]).concat(schema.views),
+            ),
+          )
+          .map((e) => {
+            const mainTableAlias = new TableAlias();
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/edit-panel/mapping-editor/InstanceSetImplementationSourceSelectorModal.tsx:145]: lets name this better than `e` maybe `relation`?

**Review by MauricioUyaguari — COMMENTED** [review]
Very minor comments. Good to approve once the changeset is addressed. Thanks!

File: .changeset/tender-foxes-battle.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-studio': patch
+---
+
+Replaced Table/View with TableAlias in MappingElementSource to fix the bug with db value for mappingTest when there are nested databases
```
Comment (MauricioUyaguari) [inline .changeset/tender-foxes-battle.md:5]:
````suggestion
Replace `Table|View` with `TableAlias` as the relational mapping source, i.e `MappingElementSource`. Fixes generating mapping test with nested databases (see [#651](https://github.com/finos/legend-studio/issues/651)] for more details).
````

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**